test/jdk/javax/swing/JMenuBar/MenuBarRTLBug.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import java.awt.BorderLayout;
+import java.awt.ComponentOrientation;
+import java.awt.Point;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.awt.Robot;
+
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import javax.swing.event.MenuEvent;
+import javax.swing.event.MenuListener;
+import javax.swing.SwingUtilities;
+
+/*
+ * @test
+ * @key headful
+ * @bug  8277369
+ * @summary Verifies arrow traversal in RTL orientation in JMenuBar
+ */
+public class MenuBarRTLBug {
+
+    static JFrame frame;
+    static JMenuBar menuBar;
+    static JMenu firstMenu;
+    static JMenuItem a;
+    static JMenuItem b;
+    static JMenu secondMenu;
+    static JMenuItem c;
+    static JMenuItem d;
+    static JMenu thirdMenu;
+    static JMenuItem e;
+    static JMenuItem f;
+    static JMenu forthMenu;
+    static JMenu fifthMenu;
+
+    static Point p;
+    static int width;
+    static int height;
+
+    static volatile boolean passed = false;
+
+    public static void main(String[] args) throws  Exception {
```

Comment:
The submitter said that he use the native L&F should not we check it in the test?

Reply:
Might as well. Added L&F iteration in the test.